### ISSUE TITLE
[BUG] Crash when encountering unknown toml

### ISSUE THREAD

**Issue by Goldziher** [hadialqattan/pycln#164]
Hi there,

We are adding rust bindings in a branch of our repository. In this branch the `pycln` tool goes ape-shit when it encounters the `cargo.toml` file:

```shell
Traceback (most recent call last):

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/bin/pycln", line 8, in <module>
    sys.exit(app())

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/lib/python3.10/site-packages/pycln/cli.py", line 157, in main
    configs = Config(

  File "<string>", line 17, in __init__

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/lib/python3.10/site-packages/pycln/utils/config.py", line 34, in __post_init__
    ParseConfigFile(file_path, self)

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/lib/python3.10/site-packages/pycln/utils/config.py", line 125, in __init__
    self.parse()

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/lib/python3.10/site-packages/pycln/utils/config.py", line 150, in parse
    getattr(self, f"_parse_{self._path.suffix.strip('.')}")()

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/lib/python3.10/site-packages/pycln/utils/config.py", line 182, in _parse_toml
    parsed_toml = toml.load(self._path)

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/lib/python3.10/site-packages/toml/decoder.py", line 134, in load
    return loads(ffile.read(), _dict, decoder)

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/lib/python3.10/site-packages/toml/decoder.py", line 511, in loads
    ret = decoder.load_line(line, currentlevel, multikey,

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/lib/python3.10/site-packages/toml/decoder.py", line 778, in load_line
    value, vtype = self.load_value(pair[1], strictly_valid)

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/lib/python3.10/site-packages/toml/decoder.py", line 880, in load_value
    return (self.load_array(v), "array")

  File "/Users/naamanhirschfeld/.cache/pre-commit/repon3uhth2r/py_env-python3/lib/python3.10/site-packages/toml/decoder.py", line 1002, in load_array
    a[b] = a[b] + ',' + a[b + 1]

IndexError: list index out of range
```

I am not able to bypass this by having pre-commit exclude the rust bindings path for the plugin, e.g. like so:

```yaml
  - repo: https://github.com/hadialqattan/pycln
    rev: v2.0.4 # Possible releases: https://github.com/hadialqattan/pycln/releases
    hooks:
      - id: pycln
        args: [--config=pyproject.toml]
        exclude: "extensions"
```

So I assume this is some toml loading in the tool.

**Comment by hadialqattan** [comment]
@Goldziher

This is weird!

(1) Pycln shouldn't open any file that's not a `.py`/`.pyi` file except for (2) and (3).
(2) It opens configuration files which must be specified explicitly via `--config`.
(3) `.gitignore` files when `--no-gitignore` wasn't used.

I tried to make Pycln raise the same error but couldn't. I've added a real `cargo.toml` file and used it as a config file.. everything was fine, so please provide more details so I can encounter the same bug.

Also, I have some questions to make sure that we're on the same page:
(1) Did you use the `cargo.toml` file as a config file and pass its name via `--config=cargo.toml`?
(2) Could you load `cargo.toml` using the [toml lib](https://pypi.org/project/toml/) that Pycln uses (because this might be a bug in the toml library + I asked you because I couldn't raise the same error myself anyway).

It would be a plus if you could try Pycln using the command line instead of using pre-commit.

Thanks a lot for reporting this issue, I'll do my best to fix it as soon as I get enough details from you. Thanks for your patience again.

**Comment by hadialqattan** [comment]
@Goldziher If you don't mind, could you provide me the `cargo.toml` file?

**Comment by Goldziher** [comment]
Its this branch - the .pre-commit.yaml has pycln commented out. Try it out --> https://github.com/starlite-api/starlite/tree/rust-bindings

**Comment by hadialqattan** [comment]
@Goldziher, apparently it's a problem with the `toml` library itself. 

The `toml` library that Pycln uses does not support [this change](https://github.com/toml-lang/toml/pull/676) until now (allowing mixed types in arrays/lists).

So, as a quick fix try to use something like this:
```
  - repo: https://github.com/hadialqattan/pycln
    rev: v2.0.4 # Possible releases: https://github.com/hadialqattan/pycln/releases
    hooks:
      - id: pycln
        args: [--all]
        exclude: "extensions"
```
(inline command line args instead of using a config file).

**Comment by hadialqattan** [comment]
Moreover, even this popular VSCode [toml parsing extension](https://marketplace.visualstudio.com/items?itemName=bungcip.better-toml) throws an error:
![image](https://user-images.githubusercontent.com/50833881/181362407-7ffa96be-29b6-4e93-b33b-edaec3a8e927.png)

**Comment by hadialqattan** [comment]
I'll try to figure out how `poetry` bypassed this issue.

**Comment by hadialqattan** [comment]
It seems like the toml library that Pycln uses does not support toml v1.0.0 which allowed the mixing of different datatypes in arrays. So I'll consider moving to `tomlkit` which supports toml v1.0.0!

**Comment by Goldziher** [comment]
I appreciate the rapid response. Lemme know when to update to the new version and I'll test.

**Comment by hadialqattan** [comment]
> I appreciate the rapid response. Lemme know when to update to the new version and I'll test.

@Goldziher 

I've released v2.1.1 so now toml v1.0.0 is supported!
